pkg/api/v1beta3/defaults.go

```diff
@@ -25,6 +25,21 @@ import (
 
 func init() {
 	api.Scheme.AddDefaultingFuncs(
+		func(obj *ReplicationController) {
+			var labels map[string]string
+			if obj.Spec.Template != nil {
+				labels = obj.Spec.Template.Labels
+			}
+			// TODO: support templates defined elsewhere when we support them in the API
+			if labels != nil {
```

Comment:
optionl nit: suggest ```len(labels) != 0``` I *think* it doesn't make a difference, and it won't hurt either way since you check things before overwriting them below-- but for consistency and on the off chance that a 0-length but not nil map is there...